python/lsst/multiprofit/fitting/fit_catalog.py

```diff
@@ -225,6 +233,56 @@ def get_suffix_y(self) -> str:
         """Get the suffix for y-axis columns."""
         return self._get_label(self.naming_scheme, self._suffix_y)
 
+    def make_catalog(self, n_rows: int, **kwargs):
+        """Make a catalog with default-initialized column values.
+
+        Parameters
+        ----------
+        n_rows
+            The number of rows to create.
+        **kwargs
+            Keyword arguments to pass to self.schema.
+
+        Returns
+        -------
+        catalog
+            The initialized catalog.
+        columns
+            The columns as returned by self.schema.
+        """
+        columns = self.schema(**kwargs)
+        keys = [column.key for column in columns]
+        prefix = self.prefix_column
+
+        idx_flag_first = keys.index("unknown_flag")
+        idx_flag_last = idx_flag_first + len(self.flag_errors)
+        dtypes = [(f'{prefix if col.key != self.column_id else ""}{col.key}', col.dtype) for col in columns]
+
+        results = Table(np.empty(n_rows, dtype=dtypes))
+        for colname in results.colnames:
+            column = results[colname]
+            dtype = column.dtype
+            if value := (
+                self.float_fill_value
+                if np.issubdtype(dtype, np.floating)
+                else (self.integer_fill_value if np.issubdtype(dtype, np.integer) else None)
+            ):
```

Comment:
In this case, you probably want to explicitly check if `value` is not None? For e.g., what if one chooses the `integer_fill_value` to be 0 instead of -1?